pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -201,13 +215,270 @@ func (a *Azure) ValidateStore(store esv1beta1.GenericStore) error {
+	}
+	// returns a CertBundle. We return CER contents of x509 certificate
+	// see: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/services/keyvault/v7.0/keyvault#CertificateBundle
+	params := keyvault.CertificateImportParameters{
+		Base64EncodedCertificate: &val,
+		Tags: map[string]*string{
+			"managed-by": pointer.String(managerLabel),
+		},
+	}
+	_, err = a.baseClient.ImportCertificate(ctx, *a.provider.VaultURL, secretName, params)
+	if err != nil {
+		return fmt.Errorf("could not import certificate %v: %w", secretName, err)
+	}
+	return nil
+}
+
+func equalKeys(newKey, oldKey keyvault.JSONWebKey) bool {
+	// checks for everything except KeyID and KeyOps
+	return newKey.Crv == oldKey.Crv &&
+		newKey.D == oldKey.D &&
+		newKey.DP == oldKey.DP &&
+		newKey.DQ == oldKey.DQ &&
+		newKey.E == oldKey.E &&
+		newKey.K == oldKey.K &&
+		newKey.Kty == oldKey.Kty &&
+		newKey.N == oldKey.N &&
+		newKey.P == oldKey.P &&
+		newKey.Q == oldKey.Q &&
+		newKey.QI == oldKey.QI &&
+		newKey.T == oldKey.T &&
+		newKey.X == oldKey.X &&
+		newKey.Y == oldKey.Y
+}
+func (a *Azure) setKeyVaultKey(ctx context.Context, secretName string, value []byte) error {
+	key, err := getKeyFromValue(value)
+	if err != nil {
+		return fmt.Errorf("could not load private key %v: %w", secretName, err)
+	}
+	jwKey, err := jwk.New(key)
+	if err != nil {
+		return fmt.Errorf("failed to generate a JWK from secret %v content: %w", secretName, err)
+	}
+	buf, err := json.Marshal(jwKey)
+	if err != nil {
+		return fmt.Errorf("error parsing key: %w", err)
+	}
+	azkey := keyvault.JSONWebKey{}
+	err = json.Unmarshal(buf, &azkey)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling key: %w", err)
+	}
+	keyFromVault, err := a.baseClient.GetKey(ctx, *a.provider.VaultURL, secretName, "")
+	ok, err := canCreate(keyFromVault.Tags, err)
+	if err != nil {
+		return fmt.Errorf("cannot get key %v: %w", secretName, err)
+	}
+	if !ok {
+		return nil
+	}
+	if keyFromVault.Key != nil && equalKeys(azkey, *keyFromVault.Key) {
+		return nil
+	}
+	params := keyvault.KeyImportParameters{
+		Key:           &azkey,
+		KeyAttributes: &keyvault.KeyAttributes{},
+		Tags: map[string]*string{
+			"managed-by": pointer.String(managerLabel),
+		},
+	}
+	_, err = a.baseClient.ImportKey(ctx, *a.provider.VaultURL, secretName, params)
+	if err != nil {
+		return fmt.Errorf("could not import key %v: %w", secretName, err)
+	}
+	return nil
+}
+
+// PushSecret stores secrets into a Key vault instance.
 func (a *Azure) PushSecret(ctx context.Context, value []byte, remoteRef esv1beta1.PushRemoteRef) error {
-	return fmt.Errorf("not implemented")
+	objectType, secretName := getObjType(esv1beta1.ExternalSecretDataRemoteRef{Key: remoteRef.GetRemoteKey()})
+	switch objectType {
+	case defaultObjType:
+		return a.setKeyVaultSecret(ctx, secretName, value)
+	case objectTypeCert:
+		return a.setKeyVaultCertificate(ctx, secretName, value)
+	case objectTypeKey:
+		return a.setKeyVaultKey(ctx, secretName, value)
```

Comment:
I think this needs a bit of docs: what data types are supported:
1. key -> EC/PKCS1/PKCS8. It does not support jwk (which is perfectly fine)
2. cert -> CERT + private key in PEM format (is both PKCS8/PKCS1)
3. secret -> everything

Reply:
Added some docs to it :)